2020-12-15-environment-variables-maintenance.md

```diff
@@ -0,0 +1,61 @@
+# Environment Variables Maintenance Proposal
+
+## The Problems
+
+### Documentation
+
+Comment by Ernesto:
+> I believe we need a policy or convention for environment variables. Right now in some projects we are using this policy: If you add a new env variable, you need to document it in both .env.sample and the RELEASE_NOTES file.
+
+### Propagation of the change
+
+It's also hard to maintain: when you pull updates you need to verify the RELEASE_NOTES and compare that with your env variables, you may miss that something changed or that something was added.
+
+### Different requirements for different environments
+
+Not all environment variables are required for development or testing. Currently we have no clear documentation regarding what is optional for development (sometimes a variable inside the .env.sample needs to be removed locally instead of leaving it blank for example).
+
+## Possible solutions
+
+Some suggestions by Ernesto and comments on Twitter were:
+
+### Envied
+
+URL: https://github.com/eval/envied
+
+Pros:
+- To use this gem we need to configure the variables in a `Envfile` (which is just ruby), and then run `ENVied.require` during the app boot.
+- We can add different requirements for different environments on a single file.
+- This can validate many formats.
+- Has a task to check the heroku config against the config file
+- Documentation can be added as comments on the Envfile
+
+Cons:
+- This does NOT support optional env variables (i.e.: variables that can be undefined but if present should have a specific format)
```

Comment:
I would add to this: 

- We need to add another file to the mix (we already have .env, .env.sample, and RELEASE_NOTES (only in OmbuShop))
- We are adding another dependency (this is part of the cons to the approach where we decide to use a new dependency)

Reply:
Oh yes, any approach for this will add a dependency, even if we add an initializer it's something with logic that we may need to update and it will required us to go over all the projects using that.